### ISSUE TITLE
Numbers represented in scientific notation should be treated as floating-point numbers

### ISSUE THREAD

**Issue by zyguan** [pingcap/tidb#2522]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
```sql
select 1e2;
```

2. What did you expect to see?
The type of the result should be `DOUBLE` according to [the number literals doc](https://dev.mysql.com/doc/refman/5.7/en/number-literals.html).

3. What did you see instead?
The result actually has a type of `DECIMAL`.

4. What version of Go are you using (`go version`)?
```
go version go1.7.4 linux/amd64
```

**Comment by zyguan** [comment]
Does following patch fix the issue properly? Do I miss something? @shenli @coocood 

```diff
diff --git a/util/types/field_type.go b/util/types/field_type.go
index e0035c9f..64167c36 100644
--- a/util/types/field_type.go
+++ b/util/types/field_type.go
@@ -134,7 +134,7 @@ func DefaultTypeForValue(value interface{}, tp *FieldType) {
                tp.Charset = mysql.DefaultCharset
                tp.Collate = mysql.DefaultCollationName
        case float64:
-               tp.Tp = mysql.TypeNewDecimal
+               tp.Tp = mysql.TypeDouble
                tp.Charset = charset.CharsetBin
                tp.Collate = charset.CharsetBin
        case []byte:
diff --git a/util/types/field_type_test.go b/util/types/field_type_test.go
index d5b99a9a..4b68e962 100644
--- a/util/types/field_type_test.go
+++ b/util/types/field_type_test.go
@@ -118,7 +118,7 @@ func (s *testFieldTypeSuite) TestDefaultTypeForValue(c *C) {
                {1, mysql.TypeLonglong},
                {uint64(1), mysql.TypeLonglong},
                {"abc", mysql.TypeVarString},
-               {1.1, mysql.TypeNewDecimal},
+               {1.1, mysql.TypeDouble},
                {[]byte("abc"), mysql.TypeBlob},
                {Bit{}, mysql.TypeBit},
                {Hex{}, mysql.TypeVarchar},
```

**Comment by coocood** [comment]
@zyguan 
No, "1.1" should be `myql.TypeNewDecimal`

**Comment by zyguan** [comment]
Hi, @coocood 
I think it's already be covered by [this](https://github.com/pingcap/tidb/pull/2461/commits/b2fe03faa3e13c5bf6173dba9f8b5018c5b95e4e#diff-6425b785337ec89d2604eb16f63caac6R53). However, #2461 failed with `An error occurred while generating the build script`. Have I done something wrong?

**Comment by coocood** [comment]
That's tranvisCI issue, I restarted it.

**Comment by zyguan** [comment]
solved by #2461
